arch/arm/src/stm32/stm32_eth.c

```diff
@@ -2046,8 +2042,23 @@ static void stm32_interrupt_work(void *arg)
       /* Clear the pending abnormal summary interrupt */
 
       stm32_putreg(ETH_DMAINT_AIS, STM32_ETH_DMASR);
+
+      /* As per the datasheet's recommendation, the MAC
+       * needs to be reset for all abnormal events. The
+       * scheduled job will take the interface down and
+       * up again.
+       */
+
+      work_queue(ETHWORK, &priv->irqwork, stm32_txtimeout_work, priv, 0);
```

Comment:
Do you need a delay to ensure this code exits with the net_unlock before stm32_txtimeout_work is called?

Reply:
Just for the reference, [here](https://github.com/apache/nuttx/pull/9064#issuecomment-1518746953) it is explained why this change is not needed.